js/stdlib/ts/ALGO.ts

```diff
@@ -2423,6 +2421,31 @@ const balanceOf = async (acc: Account | Address, token?: Token): Promise<BigNumb
   return (await balanceOfM(acc, token || null)) || bigNumberify(0);
 };
 
+const doAccountAppOptedIn = async (nacc: NetworkAccount, ctcId: BigNumber): Promise<boolean> => {
+  const ls = await getLocalState_(nacc.addr, ctcId);
+  return ls !== undefined;
+}
+const accountAppOptedIn = async (acc: Account, ctc: Contract): Promise<boolean> => {
+  return await doAccountAppOptedIn(acc.networkAccount, await ctc.getInfo());
+}
+const doAccountAppOptIn = async (nacc: NetworkAccount, ctcId: BigNumber): Promise<void> => {
```

Comment:
This function will not work for non Reach code and it is dangerous because without contract verification, the user has no idea how much it will cost